File: SQL/Select/SQLSelectExpression.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLSelectExpression
        Inherits SQLStatement

        Private pobjExpression As SQL.SQLExpression

        Public Sub New(ByVal objExpression As SQL.SQLExpression)

            If objExpression Is Nothing Then
                Throw New ArgumentNullException
            End If

            pobjExpression = objExpression

        End Sub

        Public Overrides ReadOnly Property SQL() As String
            Get

                Return "SELECT " & pobjExpression.SQL(MyBase.ConnectionType)

            End Get
        End Property

    End Class

End Namespace

File: SQL/Select/SQLSelectExpression.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLSelectExpression : SQLStatement
	{
		private SQL.SQLExpression pobjExpression;
			
		public SQLSelectExpression(SQL.SQLExpression objExpression)
		{
			if (objExpression == null)
				throw new ArgumentNullException();
				
			pobjExpression = objExpression;
		}

		public SQL.SQLExpression Expression
		{
			get
			{
				return pobjExpression;
			}
		}
			
		public override string SQL
		{
			get
			{
				return base.Serializer.SerializeSelectExpression(this);
			}
		}
	}
}
